soundata/datasets/starss2022.py

```diff
@@ -0,0 +1,645 @@
+        """The clip's event annotations
+        Returns:
+            * SpatialEvents with attributes
+                * intervals (list): list of size n np.ndarrays of shape (m, 2), with intervals
+                    (as floats) in TIME_UNITS in the form [start_time, end_time]
+                * intervals_unit (str): intervals unit, one of TIME_UNITS
+                * time_step (int, float, or None): the time-step between events
+                * elevations (list): list of size n with np.ndarrays with dtype int,
+                    indicating the elevation of the sound event per time_step.
+                * elevations_unit (str): elevations unit, one of ELEVATIONS_UNITS
+                * azimuths (list): list of size n with np.ndarrays with dtype int,
+                    indicating the azimuth of the sound event per time_step if moving
+                * azimuths_unit (str): azimuths unit, one of AZIMUTHS_UNITS
+                * distances (list): list of size n with np.ndarrays with dtype int,
+                    indicating the distance of the sound event per time_step if moving
+                * distances_unit (str): distances unit, one of DISTANCES_UNITS
+                * labels (list): list of event labels (as strings)
+                * labels_unit (str): labels unit, one of LABELS_UNITS
+                * track_number_indices (list): list of track number indices (as strings)
+                * confidence (np.ndarray or None): array of confidence values
+
+        """
+        return load_spatialevents(self.csv_path)
+
+    def to_jams(self):
+        """Get the clip's data in jams format
+        Returns:
+            jams.JAMS: the clip's data in jams format
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path, metadata=self._clip_metadata
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO, sr=24000) -> Tuple[np.ndarray, float]:
+    """Load a STARSS 2022 audio file.
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to an audio file
+        sr (int or None): sample rate for loaded audio, 24000 Hz by default.
+        If different from file's sample rate it will be resampled on load.
+        Use None to load the file using its original sample rate (24000)
+    Returns:
+        * np.ndarray - the audio signal
+        * float - The sample rate of the audio file
+    """
+    audio, sr = librosa.load(fhandle, sr=sr, mono=False)
+    return audio, sr
+
+
+@io.coerce_to_string_io
+def load_spatialevents(fhandle: TextIO, dt=0.1) -> SpatialEvents:
+    """Load a STARSS 2022 annotation file
+    Args:
+        fhandle (str or file-like): File-like object or path to
+            the sound events annotation file
+        dt (float): time step
+    Raises:
+        IOError: if txt_path doesn't exist
+    Returns:
+        SpatialEvents: sound spatial events annotation data
+    """
+
+    def _process_raw_events(raw_reader, dt):
```

Comment:
Why is this function defined as an inner function instead of an outer function?

Reply:
Could be either way. since it will only be used for our temporary `SspatialEvents` class, I decided to keep it inside. It's the same way already in the TAU SSE 2019, TAU NIGENS SSE 2020, and TAU NIGENS SSE 2021 datasets.